### PR TITLE
Added make minor-release-foundation

### PR DESCRIPTION
**Pull request by kschumy**
*Issue #, if available:*

*Description of changes:*
* Added make command to generate everything needed to start a minor release. After running `make minor-release-foundation`, a PR can be opened
* Refactor releasebranch code into its own file

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cmd/release/minor/main.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		log.Fatalf("getting the new release branch: %v", err)
+	}
+	log.Printf("Added %s to SUPPORTED_RELEASE_BRANCHES", addedReleaseBranch)
+
+	prevReleaseBranch, nextReleaseBranch := string(latestSupportedReleaseBranch), string(addedReleaseBranch)
+
+	// Adds project files new release branch
+	projectFilesAddedCount, err := projects.CreateFilesAndDirectories(prevReleaseBranch, nextReleaseBranch)
+	if err != nil {
+		log.Fatalf("at the end: %v", err)
```
Comment (danbudris) [inline cmd/release/minor/main.go:32]: typo in error message?

File: cmd/release/minor/main.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	}
+	log.Printf("Generated %d project files", projectFilesAddedCount)
+
+	// Adds release directory and sets RELEASE values to "0"
+	err = createReleaseDirectoryAndFiles(nextReleaseBranch)
+	if err != nil {
+		log.Fatalf("at the end: %v", err)
```
Comment (danbudris) [inline cmd/release/minor/main.go:39]: same as above, not sure what this message means

File: cmd/release/minor/projects/projects.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+			}
+		}
+	}
+	return nil
+}
+
+func copyFile(prevReleaseBranchFilePath, nextReleaseBranchFilePath string) error {
+	existingFile, err := os.Open(prevReleaseBranchFilePath)
+	if err != nil {
+		return err
```
Comment (danbudris) [inline cmd/release/minor/projects/projects.go:93]: name the file in the error?

File: cmd/release/minor/projects/projects.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+		if _, err = io.Copy(newFile, existingFile); err != nil {
+			return fmt.Errorf("copying files: %w", err)
+		}
+	}
+	return nil
+}
+
+func getFileCount(releaseBranch string) (int, error) {
+	pathPattern := filepath.Join(values.GetProjectPathRoot(), "*", "*", releaseBranch, "*")
+	out, err := exec.Command("bash", "-c", fmt.Sprintf("find %s -type f | wc -l", pathPattern)).Output()
```
Comment (danbudris) [inline cmd/release/minor/projects/projects.go:124]: curious why we're shelling out here instead of using Go? not an issue but just wondering

**Review by danbudris — COMMENTED** [review]
and, to confirm, you used these commands already to set up the `1.26` foundation, is that right? So this is tested and working

File: cmd/release/minor/main.go (hunk 1 above)
Comment (kschumy) [reply cmd/release/minor/main.go:32]: This was left over from me debugging 😣

File: cmd/release/minor/main.go (hunk 2 above)
Comment (kschumy) [reply cmd/release/minor/main.go:39]: same as above. I changed this in the next revision

File: cmd/release/minor/projects/projects.go (hunk 3 above)
Comment (kschumy) [reply cmd/release/minor/projects/projects.go:93]: done

File: cmd/release/minor/projects/projects.go (hunk 4 above)
Comment (kschumy) [reply cmd/release/minor/projects/projects.go:124]: It seemed easier. I'm not at all sold on this approach if you'd prefer it in Go

**Comment by kschumy** [comment]
> and, to confirm, you used these commands already to set up the `1.26` foundation, is that right? So this is tested and working

yep! It generated the initial 1.26 changes

**Pull request merged**